nomad/fsm.go

```diff
@@ -628,6 +630,36 @@ func (n *nomadFSM) applyReconcileSummaries(buf []byte, index uint64) interface{}
 	return n.reconcileQueuedAllocations(index)
 }
 
+// applyAddNodeEventType applies a node event to the set of currently-available
+// events.
+func (n *nomadFSM) applyAddNodeEventType(buf []byte, index uint64) interface{} {
+	var req structs.EmitNodeEventRequest
+	if err := structs.Decode(buf, &req); err != nil {
+		n.logger.Printf("[ERR] nomad.fsm: failed to decode EmitNodeEventREquest: %v", err)
+		return err
+	}
+
+	for nodeID, nodeEvents := range req.NodeEvents {
```

Comment:
This all should be handled in the state store. Look at something like UpsertEvals.

It should be 

```
if err := n.state.UpsertNodeEvents(index, req.NodeEvents); err != nil {
 	n.logger.Printf("[ERR] nomad.fsm: failed to upsert node events: %v", err)
 	return err
 }
```